File: CSharp/Project/Talk/SimplePeel.cs
#region The Worst of Legacy Code: Forensic Development

#endregion

using System;
using System.Data.Common;
using Talk.Required;

namespace Talk
{
    public class SimplePeel
    {
        public double CalulateTotalMortage(String name, DbConnection con)
        {
            var p = Person.LoadPersonByName(name, con);
            var home = p.GetFinancalInformation().GetHomeEquityLoan();
            var payment = p.GetFinancalInformation().GetFirstMortagePayment();

            if (home != null)
            {
                return home.GetValueOrDefault() +
                       payment.GetValueOrDefault();
            }
            if (payment != null)
            {
                return payment.GetValueOrDefault();
            }
            return 0.00;
        }
    }
}

File: EcmaScript/program/app/simplePeel.js
// The Worst of Legacy Code: Forensic Development

function calulateTotalMortage(name, db) {
    const personLoader = require('./ignored/person');
    const person = personLoader.LoadPersonByName(name, db);

    if(Boolean(person.getFinancalInformation().getHomeEquityLoan()) || person.getFinancalInformation().getHomeEquityLoan() === 0) {
        return (
            person.getFinancalInformation().getHomeEquityLoan() + 
            (Boolean(person.getFinancalInformation().getFirstMortagePayment()) 
                ? person.getFinancalInformation().getFirstMortagePayment() 
                : 0)
        );
    }
    if(Boolean(person.getFinancalInformation().getFirstMortagePayment()) || person.getFinancalInformation().getFirstMortagePayment() === 0) {
        return person.getFinancalInformation().getFirstMortagePayment();
    }
    return 0;
}

module.exports = {
    calulateTotalMortage
}